internal/driver/driver.go

```diff
@@ -526,11 +585,17 @@ func (d *Driver) newDevice(name string, protocols map[string]models.ProtocolProp
 			fmt.Sprintf("wrong device serial number, expected %s=%s, actual %s=%s", SerialNumber, psn, SerialNumber, sn), nil)
 	}
 
+	rtspUriNoCredentials := &url.URL{
+		Scheme: RtspUriScheme,
+		Host:   fmt.Sprintf("%s:%s", d.rtspHostName, d.rtspTcpPort),
+	}
+	rtspUriNoCredentials.Path = path.Join(Stream, name)
```

Comment:
I do not think you need to store an additional variable here. the regular rtspUri should be unauthenticated, and only the authenticated one is created on the fly and not stored on the device struct.

For example, create the authenticated one separatly, and use it to init the trans: `err = trans.Initialize(fdPath, rtspAuthenticatedUri.String())`